client/client.go

```diff
@@ -384,21 +389,7 @@ func (r *NotaryRepository) Publish() error {
 		if err != nil {
 			return err
 		}
-		rootKeyID := r.tufRepo.Root.Signed.Roles["root"].KeyIDs[0]
-		rootKey, ok := r.tufRepo.Root.Signed.Keys[rootKeyID]
-		if !ok {
-			return errors.New(
-				"Root was signed with %s, which is not in its list of keys.")
-		}
-		canonicalRootID, err := tufutils.CanonicalKeyID(rootKey)
-		if err != nil {
-			return err
-		}
-		rootCryptoService, err := r.KeyStoreManager.GetRootCryptoService(canonicalRootID)
-		if err != nil {
-			return err
-		}
-		root, err = r.tufRepo.SignRoot(data.DefaultExpires("root"), rootCryptoService.CryptoService)
+		root, err = r.tufRepo.SignRoot(data.DefaultExpires("root"), r.cryptoService)
```

Comment:
IRL recap:  when `tuf.Repo.SignRoot` (and then `tuf.Repo.sign`) is called, all the key IDs are validated also.  So it would fail when `SignRoot` is called instead of slightly earlier.